src/ripple/protocol/AcctRoot.h

```diff
@@ -0,0 +1,367 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2021 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_PROTOCOL_ACCT_ROOT_H_INCLUDED
+#define RIPPLE_PROTOCOL_ACCT_ROOT_H_INCLUDED
+
+#include <ripple/protocol/LedgerEntryWrapper.h>
+#include <ripple/protocol/STAccount.h>
+#include <ripple/protocol/STAmount.h>
+
+namespace ripple {
+
+template <bool Writable>
+class AcctRootImpl final : public LedgerEntryWrapper<Writable>
+{
+private:
+    using Base = LedgerEntryWrapper<Writable>;
+    using SleT = typename Base::SleT;
+    using Base::wrapped_;
+
+    // This constructor is private so only the factory functions can
+    // construct an AcctRootImpl.
+    AcctRootImpl(std::shared_ptr<SleT>&& w) : Base(std::move(w))
+    {
+    }
+
+    // Friend declarations of factory functions.
+    //
+    // For classes that contain factories we must declare the entire class
+    // as a friend unless the class declaration is visible at this point.
+    friend class ReadView;
+    friend class ApplyView;
+
+public:
+    AcctRootImpl(AcctRootImpl const&) = default;
+    AcctRootImpl(AcctRootImpl&&) = default;
+
+    AcctRootImpl&
+    operator=(AcctRootImpl const& rhs)
```

Comment:
Why is the behavior of copy constructor different from that of copy-assignment operator? If we had resorted to the compiler generated default for copy-assignment, wouldn't it do exactly the same thing?

Reply:
Good spotting!  Yes, I think you're right that the copy and move assignment operators should use `= default` in place of the explicit implementation.  Good catch.

Reply:
Excellent point, @greg7mdp.  The most recent commit applies the Rule of 0 to the `AcctRoot` wrapper.  That's particularly important because that wrapper class will be used as an example for other ledger entry wrapper implementations.  It will reduce boiler plate in all of the wrappers.

Reply:
@ckeshava, as a heads up, you should be aware that I've modified the `AcctRoot` implementation as @greg7mdp suggested.  And I'll explicitly say Thank You to both you and @greg7mdp for the improvements.